### PR TITLE
Updated lib/logger.js to ES2015

### PR DESCRIPTION
**Pull request by Nathan-Schwartz**
Tests pass and file runs in Node without Babel.

Linting passes except for the no-magic-numbers rule. I don't know what the numbers represent, so if anyone wants to help me come up with names for them I'm happy to fix the errors. Otherwise, I'm not sure if I should ignore the rule or let the linter fail for now.

### CODE REVIEW

File: lib/logger.js [hunk 1]
```diff
@@ -50,57 +52,53 @@ function trimTo(str, amt) {
   return result;
 }
 
-function Logger(cons) {
-  var logger = cons || console;
-  log = function () {
-    logger.log.apply(logger, arguments);
+function Logger(logger = console) {
```
Comment (milesj) [inline lib/logger.js]: Rewrite the `Logger` into a class and have the module export an instance.

```
class Logger {
  constructor(logger = console) {
    this.logger = logger;
  }
}
```

File: lib/logger.js [hunk 2]
```diff
@@ -50,57 +52,53 @@ function trimTo(str, amt) {
...
-      if (typeof arguments[h] === 'number') {
-        padsWidth += arguments[h];
-        pads++;
+  this.log.cols = (...args) => {
+    const width = process.stdout.columns;
+    const input = args;
+    let pads = 0;
+    let padsWidth = 0;
+
+    for (let h = 0; h < args.length; h += 1) {
```
Comment (milesj) [inline lib/logger.js]: Use `args.forEach`.

File: lib/logger.js [hunk 3]
```diff
@@ -14,15 +14,16 @@ var ut = require('util');
  */
 
 function viewed(str) {
-  var re = /\u001b\[\d+m/gm;
+  const re = /\u001b\[\d+m/gm;
```
Comment (milesj) [inline lib/logger.js]: Move the regex to a constant in the module scope, seeing as how its used multiple times.

```
const ANSI_ESCAPE_PATTERN = /\u001b\[\d+m/gm;
```

File: lib/logger.js [hunk 4]
```diff
@@ -14,15 +14,16 @@ var ut = require('util');
...
+
   return String(str).replace(re, '');
 }
 
 function trimTo(str, amt) {
```
Comment (milesj) [inline lib/logger.js]: Move this into a method on `Logger`.

File: lib/logger.js [hunk 5]
```diff
@@ -14,15 +14,16 @@ var ut = require('util');
...
-  var raw = '';
-  var visual = viewed(str).slice(0, amt);
-  var result = '';
-  for (var i = 0; i < str.length; ++i) {
+  const visual = viewed(str).slice(0, amt);
+  let raw = '';
+  let result = '';
+  for (let i = 0; i < str.length; i += 1) {
```
Comment (milesj) [inline lib/logger.js]: Add a space above.

File: lib/logger.js [hunk 6]
```diff
@@ -33,8 +34,9 @@ function trimTo(str, amt) {
     return result;
   }
 
-  var newResult = result; var found = false;
-  for (var j = result.length; j > 0; --j) {
+  let newResult = result;
+  let found = false;
+  for (let j = result.length; j > 0; j -= 1) {
```
Comment (milesj) [inline lib/logger.js]: Add a space above.

File: lib/logger.js [hunk 7]
```diff
@@ -50,57 +52,53 @@ function trimTo(str, amt) {
   return result;
 }
 
-function Logger(cons) {
-  var logger = cons || console;
-  log = function () {
-    logger.log.apply(logger, arguments);
+function Logger(logger = console) {
+  this.log = (...args) => {
+    logger.log.bind(logger, ...args);
```
Comment (milesj) [inline lib/logger.js]: `this.logger(...args)`

File: lib/logger.js [hunk 8]
```diff
@@ -1,10 +1,10 @@
+/* eslint-disable no-magic-numbers */
 /**
  * Module dependencies.
  */
 
-var _ = require('lodash');
-var util = require('./util');
-var ut = require('util');
+const util = require('./util');
+const ut = require('util');
```
Comment (milesj) [inline lib/logger.js]: Put absolutes first.

Also, can destructure? `const { inspect } = require('util');`

File: lib/logger.js [hunk 9]
```diff
@@ -50,57 +52,53 @@ function trimTo(str, amt) {
...
-    var lines = [];
+    const cols = args.length - pads;
+    const colsWidth = Math.floor((width - padsWidth) / cols);
+    const lines = [];
 
-    var go = function () {
-      var str = '';
-      var done = true;
-      for (var i = 0; i < input.length; ++i) {
+    const go = () => {
```
Comment (milesj) [inline lib/logger.js]: If you can, try and move this function into a method. Maybe something like `render(input)`.

File: lib/logger.js [hunk 10]
```diff
@@ -115,14 +113,16 @@ function Logger(cons) {
       }
     };
     go();
-    for (var i = 0; i < lines.length; ++i) {
+    for (let i = 0; i < lines.length; i += 1) {
```
Comment (milesj) [inline lib/logger.js]: `lines.forEach`

File: lib/logger.js [hunk 11]
```diff
@@ -133,4 +133,4 @@ function Logger(cons) {
  * Expose `logger`.
  */
 
-module.exports = exports = Logger;
+module.exports = Logger;
```
Comment (milesj) [inline lib/logger.js]: `module.exports = new Logger();`

File: lib/logger.js [hunk 12]
```diff
@@ -1,10 +1,10 @@
+/* eslint-disable no-magic-numbers */
```
Comment (milesj) [inline lib/logger.js]: Which number is a magic number? Probably best to disable on only that line.

File: lib/logger.js (hunk 5 above)
Comment (Nathan-Schwartz) [inline lib/logger.js]: @milesj I didn't get any trouble from the linter for this, do you think we could add a linting rule?

File: lib/logger.js (hunk 8 above)
Comment (Nathan-Schwartz) [inline lib/logger.js]: @milesj I didn't get any trouble from the linter for ordering absolute vs relative imports, do you think we could add a linting rule?

**Comment by Nathan-Schwartz** [comment]
@milesj I'll try to update the PR to address your feedback later today

File: lib/logger.js (hunk 8 above)
Comment (milesj) [inline lib/logger.js]: There are linting rules but I believe they only work for `import/export`. I'll dig around and see if there are others.

**Comment by milesj** [comment]
Closing, as I handled this myself.

**Pull request closed without merging**